server/src-lib/Hasura/Server/Auth.hs

```diff
@@ -9,42 +9,46 @@
 module Hasura.Server.Auth
   ( getUserInfo
   , AuthMode(..)
+  , AccessKey
+  , Webhook
+  , RawJWT
+  , JWTConfig (..)
+  , processJwt
   ) where
 
-import           Control.Exception        (try)
+import           Control.Exception      (try)
 import           Control.Lens
 import           Data.Aeson
-import           Data.CaseInsensitive     (CI (..), original)
-
-import qualified Data.ByteString          as B
-import qualified Data.ByteString.Lazy     as BL
-import qualified Data.HashMap.Strict      as M
-import qualified Data.Text                as T
-import qualified Data.Text.Encoding       as TE
-import qualified Data.Text.Encoding.Error as TE
-import qualified Network.HTTP.Client      as H
-import qualified Network.HTTP.Types       as N
-import qualified Network.Wreq             as Wreq
+import           Data.CaseInsensitive   (CI (..), original)
+
+import qualified Data.ByteString.Lazy   as BL
+import qualified Data.HashMap.Strict    as M
+import qualified Data.Text              as T
+import qualified Network.HTTP.Client    as H
+import qualified Network.HTTP.Types     as N
+import qualified Network.Wreq           as Wreq
 
 import           Hasura.Prelude
 import           Hasura.RQL.Types
+import           Hasura.Server.Auth.JWT
 import           Hasura.Server.Logging
+import           Hasura.Server.Utils
+
+import qualified Hasura.Logging         as L
 
-import qualified Hasura.Logging           as L
 
-bsToTxt :: B.ByteString -> T.Text
-bsToTxt = TE.decodeUtf8With TE.lenientDecode
+type AccessKey = T.Text
```

Comment:
Let's wrap them in newtypes. I've noticed couple of functions which seem to take `.. Maybe Text -> Maybe Text ..`